src/dots.jsx

```diff
@@ -25,8 +25,9 @@ export var Dots = React.createClass({
       slidesToScroll: this.props.slidesToScroll
     });
 
-    var dots = Array.apply(null, {length: dotCount}).map((x, i) => {
+    var dots = [];
 
+    for (var i=0; i<dotCount; i++) {
```

Comment:
@vramana To make ReactJS work with IE8, you have to include es5-shim anyway. Also I think it's not necessary to include the polyfill for addEventListener, it should be specified in README like ReactJS does in its doc.
http://facebook.github.io/react/docs/working-with-the-browser.html#browser-support-and-polyfills
I can send another PR for all these stuffs